source/common/listener_manager/active_tcp_socket.cc

```diff
@@ -21,14 +22,12 @@ ActiveTcpSocket::ActiveTcpSocket(ActiveStreamListenerBase& listener,
   listener_.stats_.downstream_pre_cx_active_.inc();
 
   // Automatically populate network namespace from listener address if present.
-  const auto& local_address = socket_->connectionInfoProvider().localAddress();
-  if (local_address && local_address->networkNamespace().has_value() &&
-      !local_address->networkNamespace()->empty()) {
-    stream_info_->filterState()->setData(Network::DownstreamNetworkNamespace::key(),
-                                         std::make_unique<Network::DownstreamNetworkNamespace>(
-                                             local_address->networkNamespace().value()),
-                                         StreamInfo::FilterState::StateType::ReadOnly,
-                                         StreamInfo::FilterState::LifeSpan::Connection);
+  if (network_namespace) {
```

Comment:
Hmmm, we are removing the check for empty string here. If `networkNamespace()` returns an optional containing empty string then it will now set the filter state with empty value. Is this intended?